modules/carto/src/layers/utils.ts

```diff
@@ -16,13 +16,26 @@ export function injectAccessToken(loadOptions: any, accessToken: string): void {
   }
 }
 
-export function mergeBoundaryData(geometry: VectorTile, properties: PropertiesTile): VectorTile {
+export function mergeBoundaryData(
+  geometry: VectorTile,
+  properties: PropertiesTile,
+  matchingColumn: string
+): VectorTile {
   const mapping = {};
-  for (const {geoid, ...rest} of properties.properties) {
-    if (geoid in mapping) {
-      log.warn('Duplicate geoid key in boundary mapping, using first occurance')();
+  for (const property of properties.properties) {
+    const matchingKey = property[matchingColumn];
+    if (!matchingKey || matchingKey === '') {
```

Comment:
nit, redundant falsy check:

```suggestion
    if (!matchingKey) {
```